File: client/Infrastructure Layer/Transport/src/main/java/transport/TransportContext.java
package transport;

import transport.interfaces.ITransportStrategy;

public class TransportContext {
    private ITransportStrategy strategy;

    public TransportContext(ITransportStrategy strategy) {
        this.strategy = strategy;
    }

    public void setStrategy(ITransportStrategy strategy) {
        this.strategy = strategy;
    }

    public String executeSend(String jsonToSend) {
        return strategy.sendJson(jsonToSend);
    }
}

File: server/InterfaceLayer/Transport/TransportContext.go
package transport

// TransportContext mantiene una referencia a una estrategia de transporte
type TransportContext struct {
    strategy ITransportStrategy
}

// NewTransportContext crea una nueva instancia de TransportContext
func NewTransportContext(strategy ITransportStrategy) *TransportContext {
    return &TransportContext{
        strategy: strategy,
    }
}

// SetStrategy cambia la estrategia de transporte
func (t *TransportContext) SetStrategy(strategy ITransportStrategy) {
    t.strategy = strategy
}

// ExecuteSend ejecuta el envío del JSON usando la estrategia actual
func (t *TransportContext) ExecuteSend(jsonToSend string) string {
    return t.strategy.SendJson(jsonToSend)
}
